### PR TITLE
Seri logger

### PR DESCRIPTION
**Pull request by HarryCordewener**

### CODE REVIEW

File: GuaranteedRate.Sextant.Integration.Tests/Logging/SeriLoggerIntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using GuaranteedRate.Sextant.Integration.Core;
+using GuaranteedRate.Sextant.Logging;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+
+namespace GuaranteedRate.Sextant.Integration.Tests.Logging
+{
+	[TestFixture]
+	public class SeriLoggerIntegrationTests
```
Comment (HarryCordewener) [inline GuaranteedRate.Sextant.Integration.Tests/Logging/SeriLoggerIntegrationTests.cs]: Integration Test

File: GuaranteedRate.Sextant.Tests/Logging/SeriLoggerUnitTests.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using GuaranteedRate.Sextant.Logging;
+using NUnit.Framework;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GuaranteedRate.Sextant.Tests.Logging
```
Comment (HarryCordewener) [inline GuaranteedRate.Sextant.Tests/Logging/SeriLoggerUnitTests.cs]: Unit test

File: GuaranteedRate.Sextant/Logging/Logger.cs [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 using Serilog.Formatting.Json;
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Globalization;
+using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 
 namespace GuaranteedRate.Sextant.Logging
```
Comment (HarryCordewener) [inline GuaranteedRate.Sextant/Logging/Logger.cs]: Old Class Update

File: GuaranteedRate.Sextant/Logging/SeriLogger.cs [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+using Serilog.Context;
+using Serilog.Events;
+using Serilog.Exceptions;
+using Serilog.Formatting.Json;
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace GuaranteedRate.Sextant.Logging
```
Comment (HarryCordewener) [inline GuaranteedRate.Sextant/Logging/SeriLogger.cs]: New Class

File: GuaranteedRate.Sextant/Logging/SerilogHelpers.cs [hunk 5]
```diff
@@ -1,132 +1,116 @@
...
 
-        internal static LogEventLevel GetMinLevel(IEncompassConfig config, string prefix)
-        {
-            string all_key = "";
-            string debug_key = "";
-            string info_key = "";
-            string warn_key = "";
-            string error_key = "";
-            string fatal_key = "";
+		internal static LogEventLevel GetMinLevel(IEncompassConfig config, string prefix)
```
Comment (HarryCordewener) [inline GuaranteedRate.Sextant/Logging/SerilogHelpers.cs]: Formatting

File: GuaranteedRate.Sextant/Logging/Logger.cs [hunk 6]
```diff
@@ -102,13 +107,15 @@ public static void Setup(IEncompassConfig config, Dictionary<string, string> add
                             _additionalTags.TryAdd(kv.Key, kv.Value);
                         }
                     }
-                    _additionalTags.TryAdd("process", Process.GetCurrentProcess().ProcessName);
-                    _additionalTags.TryAdd("processid", Process.GetCurrentProcess().Id.ToString());
-                    _additionalTags.TryAdd("hostname", Environment.MachineName);
-                    _additionalTags.TryAdd("windowsuser", Environment.UserName);
```
Comment (webnelly) [inline GuaranteedRate.Sextant/Logging/Logger.cs]: Do these default tags still get written to the logs somewhere else?  I've use all the time in Loggly for filtering, and if not, I think we'd have to add them to all the consuming applications individually.

File: GuaranteedRate.Sextant/Logging/SeriLogger.cs [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                try
+                {
+                    var baseLogger = new LoggerConfiguration()
+                        .WriteTo.Logger(aa => aa.MinimumLevel.Verbose())
+                        .Enrich.FromLogContext()
+                        .Enrich.WithExceptionDetails()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithProcessId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Enrich.WithProcessName();
```
Comment (webnelly) [inline GuaranteedRate.Sextant/Logging/SeriLogger.cs]: Okay, it looks like this is where those previous defaulted tags are replaced with.

File: GuaranteedRate.Sextant/Logging/Logger.cs [hunk 8]
```diff
@@ -102,13 +107,15 @@ public static void Setup(IEncompassConfig config, Dictionary<string, string> add
...
 
                     baseLogger = new LoggerConfiguration()
-                        .WriteTo.Logger(aa => aa.MinimumLevel.Verbose());
+                        .WriteTo.Logger(aa => aa.MinimumLevel.Verbose())
+                        .Enrich.FromLogContext()
+                        .Enrich.WithExceptionDetails()
+                        .Enrich.WithMachineName()
+                        .Enrich.WithProcessId()
+                        .Enrich.WithEnvironmentUserName()
+                        .Enrich.WithProcessName();
```
Comment (webnelly) [inline GuaranteedRate.Sextant/Logging/Logger.cs]: I see their replacements here.  Are these out-of-the-box methods, and if so, are there any others that we'd find useful?  Or do we need to implement those somewhere ourselves?

Comment (HarryCordewener) [reply GuaranteedRate.Sextant/Logging/Logger.cs]: image

Comment (HarryCordewener) [reply GuaranteedRate.Sextant/Logging/Logger.cs]: We can also add any properties we want as we go. I'll show you how to use Contexts later to drive a lot of awesome stuff.

File: GuaranteedRate.Sextant/Logging/SeriLogger.cs (hunk 4 above)
Comment (ratebean) [reply GuaranteedRate.Sextant/Logging/SeriLogger.cs]: Three namespaces can be removed.
image

File: GuaranteedRate.Sextant/Logging/SeriLogger.cs [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    /// We could write a wrapper around Serilog, but this isn't exactly a great project when we have a solid logging system to just make use of.
+	/// </summary>
+	public static class SeriLogger
+	{
+		private static readonly object syncRoot = new object();
+        private static bool configured = false;
+
+        public static Serilog.ILogger Logger => _logger.Value;
+
+
```
Comment (ratebean) [inline GuaranteedRate.Sextant/Logging/SeriLogger.cs]: Could remove a carriage return here

File: GuaranteedRate.Sextant/Logging/SeriLogger.cs [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+                        LogEventLevel eventLevel = SetLogglyEventLevelFromConfig(config);
+                        baseLogger.WriteTo.Loggly(logglyConfig: SerilogHelpers.GetLogglyConfig(config), formatProvider: CultureInfo.CurrentCulture,
+                                                  restrictedToMinimumLevel: eventLevel);
+                    }
+
+                    if (config.GetValue(Logging.Logger.CONSOLE_ENABLED, false))
+                    {
+                        baseLogger.WriteTo.Console(new JsonFormatter(null, false, null));
+                    }
+
```
Comment (ratebean) [inline GuaranteedRate.Sextant/Logging/SeriLogger.cs]: And here

**Pull request merged**